Source/ui_qt/bootablelistdialog.cpp

```diff
@@ -194,26 +219,43 @@ void BootableListDialog::on_awsS3Button_clicked()
 	if(bucketName.empty())
 		return;
 
-	auto getListFuture = std::async([bucketName]() {
+	m_statusBar->show();
+	auto getListFuture = std::async(std::launch::async, [this, bucketName]() {
+		m_s3_processing = true;
 		auto accessKeyId = CS3ObjectStream::CConfig::GetInstance().GetAccessKeyId();
 		auto secretAccessKey = CS3ObjectStream::CConfig::GetInstance().GetSecretAccessKey();
-		return AmazonS3Utils::GetListObjects(accessKeyId, secretAccessKey, bucketName);
-	});
-
-	auto updateBootableCallback = [this, bucketName](auto& result) {
+		AsyncUpdateStatus("Requesting S3 Bucket Content.");
+		auto result = AmazonS3Utils::GetListObjects(accessKeyId, secretAccessKey, bucketName);
+		auto size = result.objects.size();
+		int i = 1;
 		for(const auto& item : result.objects)
 		{
 			auto path = string_format("//s3/%s/%s", bucketName.c_str(), item.key.c_str());
 			try
 			{
-				TryRegisteringBootable(path);
+				std::string msg = string_format("Processing: %s (%d/%d)", path.c_str(), i, size);
+				AsyncUpdateStatus(msg);
+				if(TryRegisteringBootable(path))
+				{
+					m_bootables = BootablesDb::CClient::GetInstance().GetBootables(m_sortingMethod);
```

Comment:
Instead of potentially updating all the models on every single object, could we save a flag that would say that there was a change and call the ResetModel at the end of that lambda if something changed?

Reply:
we could, just to note, this was done like this to avoid the UI looking stuck with no new entries, im just going to rewrite it so that it updates once at the end.

but we could also add a fake entry to m_bootables, everytime we find a new entry? to keep the UI ticking?

Reply:
Yes, adding entries in the UI as they are being found by the S3 scanner would be nice, but I think it would be better to do it using a different approach. Something like firing an event when a new bootable has been registered. The UI code would be able to catch the event and add a new entry in the list view (or something else). Ideally, the S3 scanning logic should be completely UI agnostic.